src/inspect_ai/model/_providers/openai_responses.py

```diff
@@ -134,6 +154,36 @@ def model_call() -> ModelCall:
         ), model_call()
 
 
+async def wait_for_background_response(
+    client: AsyncAzureOpenAI | AsyncOpenAI, model_response: Response
+) -> Response:
+    # do some retrying so we don't waste expensive background work
+    # because of transient networking issues
+    @retry(
+        wait=wait_exponential_jitter(),
+        stop=stop_after_attempt(5) | stop_after_delay(60),
+        retry=retry_if_exception(httpx_should_retry),
+        before_sleep=log_httpx_retry_attempt(
+            f"background polling: {model_response.model}"
+        ),
+    )
+    async def check_model_response(model_response: Response) -> Response:
+        return await client.responses.retrieve(model_response.id)
+
+    try:
+        # keep checking status until we get "complete", "incomplete", of "failed"
+        while model_response.status in {"queued", "in_progress"}:
+            await anyio.sleep(5)
+            model_response = await check_model_response(model_response)
+        return model_response
+    except anyio.get_cancelled_exc_class():
+        # if the entire sample is cancelled then let the provider know
+        # so we can stop racking up token costs
+        with anyio.move_on_after(5, shield=True):
+            await client.responses.cancel(model_response.id)
```

Comment:
If you induce an exception trying to cancel, that exception (not the original) will be raised. That doesn't feel like what you want.